ambry-cloud/src/main/java/com/github/ambry/cloud/azure/StorageClient.java

```diff
@@ -423,10 +424,20 @@ protected abstract BlobServiceAsyncClient buildBlobServiceAsyncClient(HttpClient
    * @param throwable associated with this error.
    */
   private void onError(Throwable throwable) {
-    logger.error("Blob store operation failed due to exception: " + throwable.toString());
-    azureMetrics.storageClientOperationExceptionCount.inc();
-    // All retries must have been completed internally by now.
-    azureMetrics.storageClientFailureAfterRetryCount.inc();
+    boolean logError = true;
```

Comment:
Would it be simplify by:

> if (throwable instanceof BlobStorageException && ((BlobStorageException) throwable).getStatusCode() == BLOB_ALREADY_EXISTS) 
   {
      logger.debug("Blob store operation failed due to exception: " + throwable.toString());
    } else {
      logger.error("Blob store operation failed due to exception: " + throwable.toString());
      azureMetrics.storageClientOperationExceptionCount.inc();
      // All retries must have been completed internally by now.
      azureMetrics.storageClientFailureAfterRetryCount.inc();
    }

Reply:
probably, but the boolean expression is too long. I can eliminate the logError variable at most.

Reply:
Also 409 can have multiple error messages. So we should check the error msg also.
https://learn.microsoft.com/en-us/rest/api/storageservices/blob-service-error-codes

Reply:
shortened it by checked error code instead of status code